azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/cache/CacheManager.java

```diff
@@ -58,20 +65,59 @@ public Object around(ProceedingJoinPoint point) throws Throwable {
         final boolean toUseCache = StringUtils.isBlank(condition) || ExpressionUtils.evaluate(condition, invocation, true);
         final Cache<Object, Object> cache = caches.get(name);
         if (toUseCache) {
-            return getFromCache(cache, key, point);
+            log.log(Level.INFO, String.format("loading data from cache[%s.%s] on method[%s]", name, key, signature.getName()));
+            return readCache(cache, key, point);
         }
+        log.log(Level.INFO, String.format("skipping cache[%s.%s] on method[%s]", name, key, signature.getName()));
         final Object result = point.proceed();
         if (Objects.nonNull(result)) {
             cache.put(key, Optional.of(result));
         }
         return result;
     }
 
-    private Object getFromCache(Cache<Object, Object> cache, String key, ProceedingJoinPoint point) throws Throwable {
+    @Around("cacheEvict()")
+    public Object aroundCacheEvict(@Nonnull final ProceedingJoinPoint point) throws Throwable {
+        final MethodSignature signature = (MethodSignature) point.getSignature();
+        final MethodInvocation invocation = MethodInvocation.from(point);
+        final CacheEvict annotation = signature.getMethod().getAnnotation(CacheEvict.class);
+
+        final String cacheName = StringUtils.firstNonBlank(annotation.cacheName(), annotation.value());
+        final String name = ExpressionUtils.render(cacheName, invocation);
+        final String key = ExpressionUtils.render(annotation.key(), invocation);
+        final String condition = annotation.condition();
+        final boolean toEvictCache = StringUtils.isBlank(condition) || ExpressionUtils.evaluate(condition, invocation, true);
+
+        if (toEvictCache) {
+            log.log(Level.INFO, String.format("evict cache[%s.%s] on method[%s]", name, key, signature.getName()));
+            invalidateCache(name, key);
+        }
+        return point.proceed();
+    }
+
+    private void invalidateCache(@Nullable final String name, @Nullable final String key) throws ExecutionException {
+        if (StringUtils.isBlank(name)) { // invalidate all cache entries if cache name not specified
+            log.log(Level.INFO, "invalidate all cache");
+            caches.invalidateAll();
```

Comment:
Now the default value for `cacheName` is empty, which mean will invalidate all cache by default, which may be used by mistake. And I think it's a pretty dangerous operation to clear all cache, I prefer to user a keyword for all cache or provide a static method to do this job.  Same for clean all value in specific cache below.